src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -2761,6 +2757,21 @@ namespace SurfaceGeometry {
                 int const numSpaces = state.dataHeatBal->Zone(thisSurf.Zone).numSpaces;
                 int const lastSpaceForZone = state.dataHeatBal->Zone(thisSurf.Zone).spaceIndexes(numSpaces);
                 thisSurf.spaceNum = lastSpaceForZone;
+                if ((thisSurf.ExtBoundCond == unreconciledZoneSurface) && (thisSurf.Name.substr(0, 3) == "iz-")) {
+                    if (state.dataHeatBal->Zone(thisSurf.Zone).numSpaces > 1) {
+                        // Only trigger warning if the spaceless surface is an autogenerated interzone surface
+                        ShowWarningError(state,
+                                         format("{}Surface=\"{}\" has Outside Boundary Condition=Zone, but Zone=\"{}\" has more than 1 Space.",
+                                                RoutineName,
+                                                thisSurf.Name.substr(3),
+                                                thisSurf.ZoneName));
+                        ShowContinueError(state,
+                                          format("Auto-generated surface=\"{}\" will be assigned to Space=\"{}\"",
+                                                 thisSurf.Name,
+                                                 state.dataHeatBal->space(thisSurf.spaceNum).Name));
+                        ShowContinueError(state, "Use Outside Boundary Condition = Space to specify the exact Space for the outside boundary.");
+                    }
+                }
```

Comment:
After all of the spaces have been set up, every surface is already assigned to a zone, but some or all surfaces may not be assigned to a space yet. The existing logic assigns the last space in the zone. That logic remains the same here.

But if the surface was autogenerated using the Zone BC, but the other side zone has more than one space, a warning is issued, because the desired space is ambiguous.

Reply:
I think I'm tracking this.  And seems reasonable to assume that and emit a warning.